.github/workflows/main.yml

```diff
@@ -107,4 +109,104 @@ jobs:
         run: |
            chmod +x ./scripts/run_tests.sh
            ./scripts/run_tests.sh
-        shell: bash
+      - name: upload wasm plugin as artifact
+        uses: actions/upload-artifact@v2
+        if: github.ref == 'refs/heads/add-automatic-release' && contains(github.event.head_commit.message, '[release]')
+        with:
+          name: wasm-file
+          path: dist/plugin.wasm
+
+  release:
+    name: Release test
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/add-automatic-release' && contains(github.event.head_commit.message, '[release]')
+    needs: [ci]
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # fetches all history for all tags and branches
+      - uses: actions/download-artifact@v2
+        with:
+          name: wasm-file
+      - name: Bump version and create new tag
+        id: bump_version
```

Comment:
can you extract this logic to some bash script under ./scripts ?
It bloats up the action file unnecessarilly this way.